### PR TITLE
🌱  bump cluster-api to v0.3.9 

### PR DESCRIPTION
**Pull request by nader-ziada**
Co-authored-by: sedefsavas <ssavas@vmware.com>
Co-authored-by: cpanato <ctadeu@gmail.com>

 

**What this PR does / why we need it**:
- bump cluster-api to v0.3.9 
- use cluster resource set for e2e tests for e2e tests since its now required by the framework

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
update cluster-api dependency to v0.3.9 
```

### CODE REVIEW

File: templates/test/prow/cni-resource-set.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  name: ${CLUSTER_NAME}-crs-0
+  namespace: default
+spec:
+  clusterSelector:
+    matchLabels:
+      cni: ${CLUSTER_NAME}-crs-0
+  resources:
+  - kind: ConfigMap
+    name: cni-${CLUSTER_NAME}-crs-0
+  strategy: ApplyOnce
```
Comment (cpanato) [inline templates/test/prow/cni-resource-set.yaml:21]: nit: extra line

File: test/e2e/data/infrastructure-azure/cluster-template-kcp-adoption.yaml [hunk 2]
```diff
@@ -279,3 +281,24 @@ spec:
...
+  name: ${CLUSTER_NAME}-crs-0
+  namespace: default
+spec:
+  clusterSelector:
+    matchLabels:
+      cni: ${CLUSTER_NAME}-crs-0
+  resources:
+  - kind: ConfigMap
+    name: cni-${CLUSTER_NAME}-crs-0
+  strategy: ApplyOnce
```
Comment (cpanato) [inline test/e2e/data/infrastructure-azure/cluster-template-kcp-adoption.yaml:304]: nit: extra line

**Comment by CecileRobertMichon** [comment]
The upgrade failed again

```
  Running the KCP upgrade spec
  /home/prow/go/src/sigs.k8s.io/cluster-api-provider-azure/test/e2e/capi_test.go:58
    Should successfully upgrade Kubernetes, DNS, kube-proxy, and etcd in a HA cluster [It]
    /home/prow/go/pkg/mod/sigs.k8s.io/cluster-api@v0.3.9/test/e2e/kcp_upgrade.go:111
    Timed out after 1800.000s.
      
    Error: Unexpected non-nil/non-zero extra argument at index 1:
    	<*errors.fundamental>: old nodes remain
    /home/prow/go/pkg/mod/sigs.k8s.io/cluster-api@v0.3.9/test/framework/machine_helpers.go:143
```

**Pull request merged**